backstory/src/demo/java/com/google/sps/servlets/GPT2Servlet.java

```diff
@@ -0,0 +1,99 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import au.com.origma.perspectiveapi.v1alpha1.PerspectiveAPI;
+import com.google.gson.Gson;
+import com.google.sps.story.StoryManager;
+import com.google.sps.story.StoryManagerImpl;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/** Servlet that returns a generated story. */
+@WebServlet("/gpt2")
+public final class GPT2Servlet extends HttpServlet {
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json;");
+    Gson gson = new Gson();
+    String text = "";
+    String json = request.getReader().readLine();
+
+    // get the text from the JSON & handle error if it cannot be converted
+    try {
+      JSONObject jsonContainer = new JSONObject(json);
+      text = jsonContainer.getString("text");
+    } catch (JSONException exception) {
+      String errorMessage = "Could not convert text sent to server from JSON.";
+      System.out.println(text);
+
+      handleError(response, HttpServletResponse.SC_BAD_REQUEST, errorMessage);
+      return;
+    }
+
+    // Check that text is valid
+    if (text == null || text.equals("")) {
+      handleError(response, HttpServletResponse.SC_BAD_REQUEST, "Text input was null or empty");
+      return;
+    }
+    String generatedText;
+    try {
+      StoryManager storyManager = new StoryManagerImpl(text, 200, 0.7);
+      generatedText = storyManager.generateText();
+    } catch (Exception e) {
+      // Displays if internal server error.
+      handleError(response, HttpServletResponse.SC_INTERNAL_SERVER_ERROR, "Server Error");
+      return;
+    }
+
+    String jsonText = gson.toJson(generatedText);
```

Comment:
What's the difference between `jsonText` and `generatedText` here. Wouldn't these be identical, since generatedText is just a string?

Reply:
Done